File: 56. Merge Intervals.cpp
static bool comp(vector<int>& a, vector<int>& b)
{
    return a[0] < b[0] ; 
}

vector<vector<int>> merge(vector<vector<int>>& intervals) 
{
    sort(intervals.begin(), intervals.end(), comp); 

    vector<vector<int>> res ;
    res.push_back(intervals[0]); 

    for(int i = 1 ; i < intervals.size() ; i++)
    {             
        if(res.back()[1] >= intervals[i][0])
        {
            res.back()[1] = max(intervals[i][1], res.back()[1]);
        }
        else
            res.push_back(intervals[i]) ;
    }
    return res ; 
}
   

File: Java/56. Merge Intervals.java

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        List<int[]> result = new ArrayList<>();

        for (int[] vals : intervals) {
            if (result.isEmpty() || result.getLast()[1] < vals[0])
                result.add(vals);
            else
                result.getLast()[1] = Math.max(result.getLast()[1], vals[1]);
        }

        return result.toArray(int[][]::new);
    }
}